tools/reproduce_comparison/Jenkinsfile

```diff
@@ -1,56 +1,134 @@
+                    jobParams = jobParams.replace("USER_REMOTE_CONFIGS", "\"USER_REMOTE_CONFIGS\"")
+                    jobParams = jobParams.replace("ADOPT_DEFAULTS_TEMP", "\"ADOPT_DEFAULTS_JSON\"")
+                    jobParams = jobParams.replace("BUILD_CONFIGURATION", "\"BUILD_CONFIGURATION\"")
+                    
+                    def jsonJobParams = new groovy.json.JsonSlurper().parseText(jobParams)
+                    jsonJobParams.BUILD_CONFIGURATION.SCM_REF = scmRef
+                    jsonJobParams.BUILD_CONFIGURATION.BUILD_REF = buildRef
+                    jsonJobParams.BUILD_CONFIGURATION.BUILD_ARGS += " --build-reproducible-date ${buildTimeStamp}" //'${buildTimeStamp}'
+                    jsonJobParams.BUILD_CONFIGURATION.ENABLE_TESTS = "false"
+                    
+                    def buildParams = [
+                        text(name: 'BUILD_CONFIGURATION', value: "${jsonJobParams.BUILD_CONFIGURATION}"),
+                        text(name: 'USER_REMOTE_CONFIGS', value: "${jsonJobParams.USER_REMOTE_CONFIGS}"),
+                        text(name: 'DEFAULTS_JSON', value: "${jsonJobParams.DEFAULTS_JSON}"),
+                        text(name: 'ADOPT_DEFAULTS_JSON', value: "${jsonJobParams.ADOPT_DEFAULTS_JSON}")
+                    ]
+                    def reproduciableJob = build job: "${COMPARED_JOB_NAME}",
+                            propagate: false,
+                            parameters: buildParams
+                    
+                    def result = reproduciableJob.getResult()
+                    if (result == 'SUCCESS') {
+                        try {
+                            timeout(time: 2, unit: 'HOURS') {
+                                copyArtifacts(
+                                    projectName:"${COMPARED_JOB_NAME}",
+                                    selector:specific("${reproduciableJob.getNumber()}"),
+                                    filter: "**/*.tar.gz",
+                                    target: 'reproduced/',
+                                    fingerprintArtifacts: true,
+                                    flatten: true
+                                )
+                            }
+                        } catch (Exception e) {
+                            echo "Cannot run copyArtifacts from job ${COMPARED_JOB_NAME}. Exception: ${e.message}. Skipping copyArtifacts..."
                         }
-                        def retVal = sh returnStatus: true, script: "diff -q -r url1Dir1 url1Dir2 ${excludeFlags}"
-                        if (retVal != 0) {
-                            currentBuild.result = 'FAILURE'
-                            error 'Error: two builds are not the same!'
-                        } else {
-                            echo 'Success: two builds are the same!'
+                    } else {
+                        error "Reproduced build failed, exit comparision"
```

Comment:
hint to what the failed parts are?

Reply:
Add the jenkins job name. There is job link in the job console output.